### PR TITLE
Dispose FileWatcher in tests

### PR DESCRIPTION
**Pull request by BrennanConroy**
@dougbu

### CODE REVIEW

File: test/Microsoft.AspNet.StaticFiles.Tests/DefaultFilesMiddlewareTests.cs [hunk 1]
```diff
@@ -52,19 +52,22 @@ public class DefaultFilesMiddlewareTests
 
         public async Task NoMatch_PassesThrough(string baseUrl, string baseDir, string requestUrl)
         {
-            TestServer server = StaticFilesTestServer.Create(app =>
+            using (var fileProvider = new PhysicalFileProvider(Path.Combine(Directory.GetCurrentDirectory(), baseDir)))
             {
-                app.UseDefaultFiles(new DefaultFilesOptions()
+                TestServer server = StaticFilesTestServer.Create(app =>
```
Comment (dougbu) [inline test/Microsoft.AspNet.StaticFiles.Tests/DefaultFilesMiddlewareTests.cs]: `var`

File: test/Microsoft.AspNet.StaticFiles.Tests/DefaultFilesMiddlewareTests.cs [hunk 2]
```diff
@@ -88,19 +91,22 @@ public class DefaultFilesMiddlewareTests
 
         public async Task FoundDirectoryWithDefaultFile_PathModified(string baseUrl, string baseDir, string requestUrl)
         {
-            TestServer server = StaticFilesTestServer.Create(app =>
+            using (var fileProvider = new PhysicalFileProvider(Path.Combine(Directory.GetCurrentDirectory(), baseDir)))
             {
-                app.UseDefaultFiles(new DefaultFilesOptions()
+                TestServer server = StaticFilesTestServer.Create(app =>
```
Comment (dougbu) [inline test/Microsoft.AspNet.StaticFiles.Tests/DefaultFilesMiddlewareTests.cs]: `var`

File: test/Microsoft.AspNet.StaticFiles.Tests/DefaultFilesMiddlewareTests.cs [hunk 3]
```diff
@@ -124,16 +130,19 @@ public class DefaultFilesMiddlewareTests
...
             {
-                RequestPath = new PathString(baseUrl),
-                FileProvider = new PhysicalFileProvider(Path.Combine(Directory.GetCurrentDirectory(), baseDir))
-            }));
-            HttpResponseMessage response = await server.CreateRequest(requestUrl + queryString).GetAsync();
-
-            Assert.Equal(HttpStatusCode.Moved, response.StatusCode);
-            Assert.Equal(requestUrl + "/" + queryString, response.Headers.GetValues("Location").FirstOrDefault());
-            Assert.Equal(0, (await response.Content.ReadAsByteArrayAsync()).Length);
+                TestServer server = StaticFilesTestServer.Create(app => app.UseDefaultFiles(new DefaultFilesOptions()
```
Comment (dougbu) [inline test/Microsoft.AspNet.StaticFiles.Tests/DefaultFilesMiddlewareTests.cs]: `var`

File: test/Microsoft.AspNet.StaticFiles.Tests/DefaultFilesMiddlewareTests.cs [hunk 4]
```diff
@@ -124,16 +130,19 @@ public class DefaultFilesMiddlewareTests
...
+                {
+                    RequestPath = new PathString(baseUrl),
+                    FileProvider = fileProvider
+                }));
+                HttpResponseMessage response = await server.CreateRequest(requestUrl + queryString).GetAsync();
```
Comment (dougbu) [inline test/Microsoft.AspNet.StaticFiles.Tests/DefaultFilesMiddlewareTests.cs]: `var`

File: test/Microsoft.AspNet.StaticFiles.Tests/DefaultFilesMiddlewareTests.cs [hunk 5]
```diff
@@ -159,14 +168,17 @@ public class DefaultFilesMiddlewareTests
...
-                RequestPath = new PathString(baseUrl),
-                FileProvider = new PhysicalFileProvider(Path.Combine(Directory.GetCurrentDirectory(), baseDir))
-            }));
-            HttpResponseMessage response = await server.CreateRequest(requestUrl).GetAsync();
+                TestServer server = StaticFilesTestServer.Create(app => app.UseDefaultFiles(new DefaultFilesOptions()
+                {
+                    RequestPath = new PathString(baseUrl),
+                    FileProvider = fileProvder
+                }));
+                HttpResponseMessage response = await server.CreateRequest(requestUrl).GetAsync();
```
Comment (dougbu) [inline test/Microsoft.AspNet.StaticFiles.Tests/DefaultFilesMiddlewareTests.cs]: So much `var`

Comment (BrennanConroy) [inline test/Microsoft.AspNet.StaticFiles.Tests/DefaultFilesMiddlewareTests.cs]: Those weren't even from me!

Comment (dougbu) [inline test/Microsoft.AspNet.StaticFiles.Tests/DefaultFilesMiddlewareTests.cs]: Until I hit the `?w=1` button in the URL, it all looked like you!

Comment (dougbu) [inline test/Microsoft.AspNet.StaticFiles.Tests/DefaultFilesMiddlewareTests.cs]: ... but should still clean up the code while in the neighbourhood.

**Comment by dougbu** [comment]
:shipit:

**Pull request merged**